homeassistant/components/yeelight/light.py

```diff
@@ -290,8 +291,8 @@ def _lights_setup_helper(klass):
         )
 
     hass.data[PLATFORM_DATA_KEY] += lights
-    add_entities(lights, True)
-    setup_services(hass)
+    async_add_entities(lights, True)
```

Comment:
Why do we call update during entity addition? Won't that make update be called twice?

Reply:
Currently the `_hs` and `_effect` attributes are set by `update`. `update` itself only updates attributes based on the data we have so it does not pull data from the device.